xbmc/interfaces/legacy/Player.cpp

```diff
@@ -350,6 +350,16 @@ namespace XBMCAddon
       return g_application.CurrentFileItem().GetDynPath();
     }
 
+    XBMCAddon::xbmcgui::ListItem* Player::getPlayingItem()
+    {
+      XBMC_TRACE;
+      if (!g_application.GetAppPlayer().IsPlaying())
+        throw PlayerException("XBMC is not playing any item");
+
+      CFileItemPtr itemPtr = std::make_shared<CFileItem>(g_application.CurrentFileItem());
```

Comment:
I've looked into this myself in the past as there are loads of reports related to wrong information in json's Player.GetItem. Shouldn't the item be obtained from the actual player ("proxied" by ApplicationPlayer) instead of from the Application itself? It seems every player gets the item on the `OpenFile` call, they may or may not store it as a member variable and can also do any sort of modifications to the item. If we are requesting the item from the player, it makes sense to return a copy of the "most up to date item" and not the one in application. Even if this is only implemented where actually needed now for mediaimport (e.g. in VideoPlayer) and left unimplemented on other players

Reply:
TBH I simply copied the logic from `Player::getPlayingFile()` which uses `g_application.CurrentFileItem().GetDynPath()`. Is there already an API available from the players or would it have to be added? I'm not very fond of modifying stuff in `CVideoPlayer` etc.

Reply:
You'll always have to use ApplicationPlayer for that afaics, but then ApplicationPlayer gets it from the actual player.